bayesian/api_v1.py

```diff
@@ -1076,6 +1080,61 @@ def get(runtime):
         return response, 200
 
 
+class EmptyBooster(ResourceWithSchema):
+    """Implementation of /empty-booster POST REST API call."""
+
+    method_decorators = [login_required]
+
+    @staticmethod
+    def post():
+        """Handle the POST REST API request."""
+        pom_template = fetch_file_from_github(CORE_DEPENDENCIES_REPO_URL, 'pom.template.xml')
+        remote_repo = request.form.get('gitRepository')
+        if not remote_repo:
+            raise HTTPError(400, error="Expected gitRepository in request")
+
+        dependencies = [dict(zip(['groupId', 'artifactId', 'version'],
+                                 d.split(':'))) for d in request.form.getlist('dependency')]
+
+        git_org = request.form.get('gitOrganization')
+        github_token = get_access_token('github')
+
+        if pom_template:
+            pom_template = pom_template[0].get('content', '')
+            maven_obj = MavenPom(pom_template)
+            maven_obj.add_dependencies(dependencies)
+            dir_struct = {
+                'name': 'booster',
+                'type': 'dir',
+                'contains': [{'name': 'src',
+                              'type': 'dir',
+                              'contains': [
+                                  {'name': 'main/java/io/openshift/booster',
+                                   'type': 'dir',
+                                   'contains': {'name': 'Booster.java',
+                                                'contains': 'package io.openshift.booster;\
+                                                        \n\npublic class Booster { } '}
+                                   },
+                                  {'name': 'test/java/io/openshift/booster',
+                                   'type': 'dir',
+                                   'contains': {'name': 'BoosterTest.java',
+                                                'contains': 'package io.openshift.booster;\
+                                                        \n\npublic class BoosterTest { } '}
+                                   }]},
+                             {'name': 'pom.xml',
+                              'type': 'file',
+                              'contains': MavenPom.tostring(maven_obj)}
+                             ]
+            }
+            booster_dir = tempfile.TemporaryDirectory().name
+            create_directory_structure(booster_dir, dir_struct)
+            push_repo(github_token, os.path.join(booster_dir, dir_struct.get('name')),
+                      remote_repo, organization=git_org, auto_remove=True)
+            return {'status': 'ok'}, 200
+        else:
+            raise HTTPError(500, "Uable to fetch pom template file.")
```

Comment:
the function is ok, but for testing it might be better to refactor it into three parts: getting data from request, process data, generate response (nothing urgent ATM, you decide..)